pkg/agent/controller/agent.go

```diff
@@ -78,23 +78,31 @@ func (a *Controller) Run(stopCh <-chan struct{}) error {
 	return nil
 }
 
-func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) runtime.Object {
+func (a *Controller) serviceExportToRemoteMcs(obj runtime.Object) (runtime.Object, bool) {
 	svcExport := obj.(*lighthousev2a1.ServiceExport)
 	svc, err := a.kubeClientSet.CoreV1().Services(svcExport.Namespace).Get(svcExport.Name, metav1.GetOptions{})
 	if err != nil {
 		klog.Errorf("No matching service for %v", svcExport)
-		return nil
+		return nil, true
 	}
-	mcs := &lighthousev1.MultiClusterService{
+
+	serviceImport := &lighthousev2a1.ServiceImport{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: svc.Name + "-" + svc.Namespace + "-" + a.clusterID,
 			Annotations: map[string]string{
 				"origin-name":      svc.Name,
 				"origin-namespace": svc.Namespace,
 			},
 		},
-		Spec: lighthousev1.MultiClusterServiceSpec{
-			Items: []lighthousev1.ClusterServiceInfo{
+		Spec: lighthousev2a1.ServiceImportSpec{
+			Ports:                 nil,
+			IP:                    "10.10.10.10",
```

Comment:
Is this because this is a required field? This should be optional since it will be empty for Headless service.

Reply:
oh it was a code added to debug,, will remove it.